### ISSUE TITLE
Vault write /<auth>/tune command gives success message without actions performed

### ISSUE THREAD

**Issue by jhnlambert** [hashicorp/vault#11302]
**Describe the bug**
Using the command "vault write sys/mounts/auth/<auth type>/tune -key=value" results in the message "Success! Data written to: sys/mounts/auth/<auth type>/tune" but it doesn't write any data to the tune config.

**To Reproduce**
Steps to reproduce the behavior, using approle as the example auth method:
1. Run `vault read sys/mounts/auth/approle/tune`  confirms path and shows current config, in this example max-lease-ttl is set to "9553h"
2. Run `vault write sys/mounts/auth/approle/tune -max_lease_ttl="9552h"`
3. See message "Success! Data written to: sys/mounts/auth/approle/tune"
4. Run `vault read sys/mounts/auth/approle/tune` Shows that key "max_lease_ttl" was not updated
5. run `vault auth tune -max-lease-ttl="9552h" approle/` this command will work to update the tune config
6. Run `vault read sys/mounts/auth/approle/tune` Shows that key "max_lease_ttl" was updated by the tune command, also shows permissions are correct for ability to tune auth method

**Expected behavior**
Using the command "vault write sys/mounts/auth/<auth type>/tune -key=value" results in an error or message saying to use the tune command, or the command can update the tune. 

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.6.3
* Vault CLI Version (retrieve with `vault version`): 1.7.0
* Server Operating System/Architecture: Windows 10 build 18363 - cmd used for commands 

Vault server configuration file(s):

```hcl
#policy for updating tune
path "sys/mounts/auth/+/tune" {
  capabilities = ["read", "update"]
}
```

**Additional context**
reviewed existing issues and I couldn't find a similar one, apologies if this is a duplicate.

**Comment by kalafut** [comment]
In step 2, if you remove the hyphen and use `max_lease_ttl="9552h"` it should update the value correctly.

The hyphenated options apply more generally to many API requests (see `vault write -h` for that list), whereas in this case  `max_lease_ttl` is a data parameter to the backend (https://www.vaultproject.io/docs/commands/write)

**Comment by jhnlambert** [comment]
Hi Jim,

Thanks for the reply! 

Tested it out and I've confirmed using the command and it does update the value as expected -  `vault write sys/mounts/auth/approle/tune max_lease_ttl="9551h"`

So that takes care of the cause of why this wasn't updating. Thank you! 


It does appear that it will only take/apply data for the fields that already exist under the tune path, while still giving a success for commands to write fields that are not existing. Is that expected/normal behavior? 

IE: `vault write sys/mounts/auth/approle/tune -wrong-field="test" max_lease_ttl="9553h"` will update the max_lease_ttl but there is no "wrong-field" key that will get updated. Even if the command only has incorrect fields a success message is given.
